# Project: Microarray Leukemia Classification

### Overview
Classified microarray samples as Acute Lymphoblastic Leukemia (ALL) or Acute Myeloid Leukemia (AML) using a support vector classifier (SVC) on a small subset of the normalized intensity values of 7,129 genes from microarray samples generated by Golub, et al.
Explored effect of data preprocessing on final model, resulting in models with accuracy > 85% on test set.

### Language
Python

### Packages Used
numpy, pandas, scipy, sklearn, matplotlib, seaborn

### Data Pre-processing, 2 options
- no additional pre-processing
- zero cutoff for gene expression, log base 2 transformed (files appended log2)

### Feature Selection
Reduced 7,129 features to ~5 features for classification
- generated metrics on each gene/feature (i.e. t-test p-values, mutual information scores) 
- filtered features based on chosen threshold values for metrics
- generated correlation matrix on remaining features
- eliminated features correlated above threshold, keeping features with lower p-value or mutual information metric

### Model
Support Vector Classifier

### Observations
- The histogram of p-values for each gene indicates a high number of differentially expressed genes.
- Estimated ~50% of genes differentially expressed.
- Simulation of picking a random 5 genes for classification showed a mean classification score of ~0.62.

When data log transformed with zero cutoff for expression (files appended log2):
- Adjusted p-values obtained by permutation of the samples resulted in a slightly lower number of genes with low p-values, but the overall distribution appeared unchanged.
- Although a final feature set of ~5 genes was chosen, these may not have been the best features. A simulation of randomly picking 5 genes from the filtered gene set (before eliminating genes based on correlation) shows that another set of 5 genes could have better classified the test data.


When data used without additional preprocessing:
- Adjusted p-values obtained by permutation of the samples resulted in a slightly larger number of genes with low p-values, but the overall distribution appeared unchanged.
- Although a final feature set of ~5 genes was chosen, the classification results were comparable when some genes in the final feature set were substituted with 
highly correlated alternate genes (those eliminated in the final step of feature selection due to high correlation). 
- Since many genes are differentially expressed between the ALL and AML groups, the genes in the final model appear to be interchangable with multiple other 
differentially expressed genes for classification.


### Resources
[Kaggle](https://www.kaggle.com/)
